components/layout/Layout.tsx

```diff
@@ -131,12 +131,43 @@ export const Layout: FC<LayoutProps> = ({ children }) => {
 
   // register to phone island events
 
-  useEventListener('phone-island-main-presence', (data) => {
+  const auth = useSelector((state: RootState) => state.authentication)
+  const user = auth.username
+  const { mainPresence } = useSelector((state: RootState) => state.user)
+  const topBarPresence = mainPresence
+
+  useEventListener('phone-island-main-presence', (data: any) => {
     const opName = Object.keys(data)[0]
     const mainPresence = data[opName].mainPresence
     store.dispatch.operators.updateMainPresence(opName, mainPresence)
+    if (
+      // (data[user] && !data[user].mainPresence ) ||
+      data[user] &&
+      data[user].mainPresence !== topBarPresence
+    ) {
+      setUpdatedPresence(data[user].mainPresence)
+    }
   })
 
+  const setUpdatedPresence = async (mainPresence: any) => {
+    try {
+      const userInfo = await getUserInfo()
```

Comment:
There is non need to call `getUserInfo()`, logged user data is already inside `user` store